mesh/mesh.hpp

```diff
@@ -1529,21 +1529,44 @@ class Mesh
    /// @note The returned object is owned by the class and is shared, i.e.,
    /// calling this function resets pointers obtained from previous calls.
    /// Also, this pointer should NOT be deleted by the caller.
-   virtual FaceElementTransformations *GetFaceElementTransformations(
-      int FaceNo,
-      int mask = 31);
+   virtual FaceElementTransformations *
+   GetFaceElementTransformations(int FaceNo, int mask = 31);
+
+   /// Variant of GetFaceElementTransformations using a user allocated
+   /// FaceElementTransformations object.
+   virtual void GetFaceElementTransformations(int FaceNo,
+                                              FaceElementTransformations *FElTr,
+                                              IsoparametricTransformation *ElTr1,
+                                              IsoparametricTransformation *ElTr2,
+                                              int mask = 31) const;
```

Comment:
Personally I would uses references here, since a null pointer will segfault (and similarly for the other variants below).

Reply:
I agree, but wanted to be consistent with the `Mesh::GetElementTransformation(int i, IsoparametricTransformation *ElTr);` definition. Can I deprecate the pointer interface there and add an override with references?

Reply:
Or we just accept the inconsistent interface for now (and make note to try to make things more consistent in MFEM 5.0). No strong opinions on this one

Reply:
Done in e26f72da.